MapboxDirections/MBComponent.swift

```diff
@@ -1,45 +1,16 @@
 import Foundation
-
 /**
-The parent component that comprises what should be displayed in the instruction banner.
+The component protocol that comprises what the instruction banner should display.
  */
 @objc(MBComponent)
-open class Component: NSObject, NSSecureCoding {
-    
-    open static var supportsSecureCoding: Bool = true
-    
+public protocol Component: class, NSSecureCoding {
     /**
      The plain text representation of this component.
      */
-    @objc public let text: String?
+    @objc var text: String? { get }
```

Comment:
This property only makes sense for VisualInstructionComponent, not for LaneIndicationComponent, so remove it from this protocol (but keep it as part of VisualInstructionComponent).